iree/turbine/kernel/ops/wave_ops.py

```diff
@@ -409,11 +409,15 @@ def copy(
             new_node.reduction_dim = self.fx_node.reduction_dim
         return get_custom(new_node)
 
-    def replace_all_uses_with(self, new_node: CustomOp | fx.Node):
+    def replace_all_uses_with(
+        self,
+        new_node: CustomOp | fx.Node,
+        filter_fn: Callable[[fx.node], bool] = lambda x: True,
+    ):
```

Comment:
Can you add a comment describing what the filter function means? Does it mean that we will replace all uses except the ones that the filter function returns false for?

Reply:
Ah, this is dead code, Used this previously to make `replace_all_uses_with` work more similar to `replace_all_uses_except` where the `filter_fn` is the lambda that skips over op if it's true.

Reply:
Yes that's what it looked like to me when I first saw it. Sounds like a useful function to have but we can add that later.